src/api/fetchWithAuth.ts

```diff
@@ -0,0 +1,49 @@
+const generateBasicToken = (userId: string, userPassword: string): string => {
+  const token = btoa(`${userId}:${userPassword}`);
+  return `Basic ${token}`;
+};
+
+const API_URL = `${import.meta.env.VITE_API_URL}`;
+const USER_ID = `${import.meta.env.VITE_USER_ID}`;
+const USER_PASSWORD = `${import.meta.env.VITE_USER_PASSWORD}`;
+
+if (!API_URL || !USER_ID || !USER_PASSWORD) {
+  throw new Error(
+    "API_URL, USER_ID, PASSWORD environment variables are not set"
+  );
+}
+
+type HttpMethod = "GET" | "POST" | "PATCH" | "DELETE";
+
+interface RequestOptions {
+  method: HttpMethod;
+  body?: Record<string, unknown>;
+}
+
+export const fetchWithAuth = async (path: string, options: RequestOptions) => {
+  const requestInit = requestBuilder(options);
+  const response = await fetch(path, requestInit);
+
+  if (!response.ok) {
+    throw new Error(`Failed to ${options.method} ${path}`);
+  }
+
+  return response;
+};
+
+const requestBuilder = (options: RequestOptions): RequestInit => {
+  const token = generateBasicToken(USER_ID, USER_PASSWORD);
+
+  const { method, body } = options;
+
+  const headers: HeadersInit = {
+    "Content-Type": "application/json",
+    Authorization: token,
+  };
+
+  return {
+    method,
+    headers,
+    body: body ? JSON.stringify(body) : undefined,
+  };
+};
```

Comment:
처음부터 구조분해할당을 이용해서 parameter를 받아와도 좋을거 같네요
```suggestion
const requestBuilder = ({ method, body }: RequestOptions): RequestInit => {
  const token = generateBasicToken(USER_ID, USER_PASSWORD);

  const headers: HeadersInit = {
    "Content-Type": "application/json",
    Authorization: token,
  };

  return {
    method,
    headers,
    body: body ? JSON.stringify(body) : undefined,
  };
};

```

Reply:
오! 간결해져서 보기 훨씬 좋네요! 적극 반영하겠습니다!